ocs_ci/utility/powernodes.py

```diff
@@ -0,0 +1,132 @@
+import logging
+
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.exceptions import UnexpectedBehaviour
+from ocs_ci.ocs.node import wait_for_nodes_status, get_worker_nodes, get_master_nodes
+from ocs_ci.ocs.ocp import wait_for_cluster_connectivity
+from ocs_ci.utility.utils import TimeoutSampler, exec_cmd
+from ocs_ci.utility.utils import get_ocp_version
+
+logger = logging.getLogger(__name__)
+
+
+class POWERNODES(object):
+    """
+    Wrapper for PowerNodes
+    """
+
+    isKVM = False
+
+    def __init__(self):
+        """
+        Class Initialization.
+        Run lsmod command to prepare isKVM variable.
+        """
+        cmd = "sudo /usr/sbin/lsmod"
+        result = exec_cmd(cmd)
+        if b"kvm" in result.stdout.lower():
+            self.isKVM = True
+        else:
+            self.isKVM = False
+        logger.info(f"iskvm check: {self.isKVM}")
+
+    def iskvm(self):
+        """
+        Verify if this environment is libvirt or kvm.
+        Args: None
+        Returns:
+            bool: True if this is kvm environment, False otherwise
+        """
+        return self.isKVM
+
+    def verify_machine_is_down(self, node):
+        """
+        Verify if PowerNode is completely powered off
+
+        Args:
+            node (object): Node objects
+
+        Returns:
+            bool: True if machine is down, False otherwise
+
+        """
+        result = exec_cmd(
+            "sudo virsh domstate test-ocp" + get_ocp_version("-") + f"-{node.name}"
+        )
+        if result.stdout.lower().rstrip() == b"running":
+            return False
+        else:
+            return True
+
+    def stop_powernodes_machines(self, powernode_machines, force=True):
+        """
+        Stop PowerNode Machines
+
+        Args:
+            powernode_machines (list): PowerNode objects
+            force (bool): True for PowerNode ungraceful power off, False for
+                graceful PowerNode shutdown - for future use
+
+        Raises:
+            UnexpectedBehaviour: If PowerNode machine is still up
+
+        """
+        ocpversion = get_ocp_version("-")
+        for node in powernode_machines:
+            cmd = "sudo virsh shutdown test-ocp" + ocpversion + f"-{node.name}"
```

Comment:
We shouldn't be using concatenation and f-strings together like this. Let's simplify and make the whole thing an f-string.
 
```suggestion
            cmd = f"sudo virsh shutdown test-ocp{ocpversion}-{node.name}"
```

Reply:
Done.